tests/unit/inventory/test_nb_inventory.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2020, Hillsong, Douglas Heriot (@DouglasHeriot) <douglas.heriot@hillsong.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+import pytest
+import os
+from functools import partial
+from unittest.mock import patch, MagicMock, Mock
+
+try:
+    from ansible_collections.netbox.netbox.plugins.inventory.nb_inventory import (
+        InventoryModule,
+    )
+    from ansible_collections.netbox.netbox.tests.test_data import load_test_data
+
+except ImportError:
+    import sys
+
+    # Not installed as a collection
+    # Try importing relative to root directory of this ansible_modules project
+
+    sys.path.append("plugins/inventory")
+    sys.path.append("tests")
+    from nb_inventory import InventoryModule
+    from test_data import load_test_data
+
+load_relative_test_data = partial(
+    load_test_data, os.path.dirname(os.path.abspath(__file__))
+)
+
+
+@pytest.fixture
+def inventory_fixture():
+    # TODO: Mock _fetch_information() to return static HTTP responses
+    inventory = InventoryModule()
+    inventory.api_endpoint = "https://netbox.test.endpoint:1234"
+    return inventory
+
+
+@pytest.fixture
```

Comment:
Should we test against the actual parameters provided in the inventory file here?

Reply:
I suppose there's no reason not to. Just thought I'd make the tests simpler.
I've now changed this test fixture to get the list of device parameters.

The tests for `refresh_url` are already against the actual lists and test both device and query parameters.